### PR TITLE
libselinux: fix build with musl

### LINKED ISSUE #119471: libselinux does not build with pkgsMusl
**Describe the bug**
pkgsMusl.libselinux does not build because of:

```
builder for '/nix/store/bw1k61ngsnhqa9lz2rvrk0y05rka0pkf-libselinux-3.0.drv' failed with exit code 2; last 10 log lines:
  procattr.c:38:14: error: static declaration of ‘gettid’ follows non-static declaration
     38 | static pid_t gettid(void)
        |              ^~~~~~
  In file included from procattr.c:2:
  /nix/store/k515dcqzlwncg8flh97l9dq3vwxn7jvl-musl-1.2.2-dev/include/unistd.h:194:7: note: previous declaration of ‘gettid’ was here
    194 | pid_t gettid(void);
        |       ^~~~~~
  make[1]: *** [Makefile:158: procattr.o] Error 1
  make[1]: Leaving directory '/build/libselinux-3.0/src'
  make: *** [Makefile:44: all] Error 1
[0 built (1 failed)]
error: build of '/nix/store/bw1k61ngsnhqa9lz2rvrk0y05rka0pkf-libselinux-3.0.drv' failed
```

**To Reproduce**
1. `nix repl`
2. `:l https://github.com/NixOS/nixpkgs/archive/6802c2e2ce9ef37976a306ada908e121409093ad.tar.gz`
2. `:b pkgsMusl.libselinux`

**Expected behavior**
The library is built

**Screenshots**

**Additional context**

**Notify maintainers**

@Phreedom 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.108, NixOS, 20.09.3777.ad7604ddbd9 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(kpg): `""`
 - channels(root): `"nixos-20.09.3777.ad7604ddbd9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by Uthar**
fix #119471

### CODE REVIEW

File: pkgs/os-specific/linux/libselinux/default.nix [hunk 1]
```diff
@@ -43,6 +43,10 @@ stdenv.mkDerivation rec {
     "LIBSEPOLA=${lib.getLib libsepol}/lib/libsepol.a"
   ];
 
+  prePatch = lib.optional stdenv.hostPlatform.isMusl ''
+    sed -i 's,#include <unistd.h>,,g' src/procattr.c
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libselinux/default.nix:47]: Please use subsituteInPlace.

Comment (Uthar) [reply pkgs/os-specific/linux/libselinux/default.nix]: done in 07b190c8bdf883ae899f53d56d7f6bfea6a12cd8

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119472` at 118a3a29 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>845 packages marked as broken and skipped:</summary>
  <ul>
    <li>OVMF-CSM</li>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>aldor</li>
    <li>alliance</li>
    <li>ams-lv2</li>
    <li>amuleDaemon</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>9487 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>CHOWTapeModel</li>
    <li>DisnixWebService</li>
    <li>EBTKS</li>
    <li>EmptyEpsilon</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>MMA</li>
    <li>OVMF</li>
    <li>OVMF-secureBoot</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>27 packages built successfully:</summary>
  <ul>
    <li>avahi</li>
    <li>cups</li>
    <li>dbus</li>
    <li>dbus-glib</li>
    <li>dconf (gnome3.dconf ,xfce.dconf)</li>
    <li>desktop-file-utils</li>
    <li>fontconfig</li>
    <li>fontforge (fontforge-fonttools ,python38Packages.fontforge)</li>
    <li>gd</li>
    <li>glib (gnome2.glib ,gnome3.glib)</li>
    <li>vala (gnome3.vala ,vala_0_48)</li>
    <li>gnomeExtensions.tilingnome</li>
    <li>gts</li>
    <li>libudev0-shim</li>
    <li>libva-minimal</li>
    <li>media-player-info</li>
    <li>pcsclite</li>
    <li>python38Packages.libselinux</li>
    <li>python38Packages.pyudev</li>
    <li>python39Packages.libselinux</li>
    <li>shared-mime-info</li>
    <li>spideroak</li>
    <li>udev (systemd)</li>
    <li>wayland</li>
    <li>wayland-protocols</li>
    <li>xlibsWrapper (xorg.xlibsWrapper)</li>
    <li>xorg.libXft</li>
  </ul>
</details>

File: pkgs/os-specific/linux/libselinux/default.nix [hunk 2]
```diff
@@ -43,6 +43,11 @@ stdenv.mkDerivation rec {
     "LIBSEPOLA=${lib.getLib libsepol}/lib/libsepol.a"
   ];
 
+  prePatch = lib.optional stdenv.hostPlatform.isMusl ''
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/libselinux/default.nix:46]:
```suggestion
  postPatch = lib.optionalString stdenv.hostPlatform.isMusl ''
```

**Comment by SuperSandro2000** [comment]
@Uthar can you fix the merge conflict?

**Comment by Uthar** [comment]
> @Uthar can you fix the merge conflict?

looks good now

**Comment by Uthar** [comment]
I just realized, on master it''s still broken

**Pull request merged**